javaslang/src/main/java/javaslang/collection/Stack.java

```diff
@@ -38,7 +39,7 @@
  * @author Daniel Dietrich
  * @since 2.0.0
  */
-public interface Stack<T> extends LinearSeq<T> {
+public interface Stack<T> extends Kind1<List<?>, T>, LinearSeq<T> {
```

Comment:
haha, I was also tempted to remove it entirely... but we will go with a thin interface.